cli/internal/run/global_hash.go

```diff
@@ -23,43 +23,36 @@ var _defaultEnvVars = []string{
 	"VERCEL_ANALYTICS_ID",
 }
 
-func calculateGlobalHash(rootpath turbopath.AbsolutePath, rootPackageJSON *fs.PackageJSON, pipeline fs.Pipeline, externalGlobalDependencies []string, packageManager *packagemanager.PackageManager, logger hclog.Logger, env []string) (string, error) {
-	// Calculate the global hash
-	globalDeps := make(util.Set)
-
+func calculateGlobalHash(rootpath turbopath.AbsolutePath, rootPackageJSON *fs.PackageJSON, pipeline fs.Pipeline, envVarDependencies []string, globalFileDependencies []string, packageManager *packagemanager.PackageManager, logger hclog.Logger, env []string) (string, error) {
```

Comment:
I don't love that a new argument was added, would it be a good/bad idea to pass in the full `turboJSON` representation or does idiomatic Go prefer more arguments over unnecessary data passing?

Reply:
I love that there is a new argument here, it's super clear that it's one of the inputs to the function. 🚀